src/Test/TestAIConstraints.java

```diff
@@ -82,4 +86,30 @@ private boolean testUnmovedFriend(AIMaker ai)
 
     return testPassed;
   }
+
+  private boolean testBuildTooManyMans(AIMaker ai)
+  {
+    setupTest(TestRange.getMapInfo(), ai, new GameScenario(new AWBWUnits(), 1000, 1000, 1, FogMode.OFF_DOR, TagMode.OFF));
+    final Army armyOne = testGame.armies[0];
+    armyOne.team = 9;
+    testGame.armies[1].team = armyOne.team;
+    // Run through a round of init-turns so that our allies are ready-to-act
+    day(testGame);
```

Comment:
Is this test acting on allies at all? It looks like it just lets `armyOne` act until it's done with its first turn.